controllers/keystoneapi_controller.go

```diff
@@ -452,59 +451,89 @@ func (r *KeystoneAPIReconciler) reconcileInit(
 	// run keystone db sync - end
 
 	//
-	// expose the service (create service, route and return the created endpoint URLs)
+	// create service/s
 	//
-	var keystonePorts = map[endpoint.Endpoint]endpoint.Data{
-		endpoint.EndpointPublic: {
+	var keystoneEndpoints = map[service.Endpoint]endpoint.Data{
+		service.EndpointPublic: {
 			Port: keystone.KeystonePublicPort,
 		},
-		endpoint.EndpointInternal: {
+		service.EndpointInternal: {
 			Port: keystone.KeystoneInternalPort,
 		},
 	}
+	apiEndpoints := make(map[string]string)
 
-	for _, metallbcfg := range instance.Spec.ExternalEndpoints {
-		portCfg := keystonePorts[metallbcfg.Endpoint]
-		portCfg.MetalLB = &endpoint.MetalLBData{
-			IPAddressPool:   metallbcfg.IPAddressPool,
-			SharedIP:        metallbcfg.SharedIP,
-			SharedIPKey:     metallbcfg.SharedIPKey,
-			LoadBalancerIPs: metallbcfg.LoadBalancerIPs,
+	for endpointType, data := range keystoneEndpoints {
+		endpointName := keystone.ServiceName + "-" + string(endpointType)
```

Comment:
This feels duplicated information between here and https://github.com/openstack-k8s-operators/openstack-operator/pull/416/files#diff-dc49c70e72e8b711bfefebe68ede374400ee1c2c0fe24c4a25a1a6c352a1a489R128

Reply:
I initially moved keystone.ServiceName to be defined in the api module. we can have it there and use it in both places?

Reply:
let's discuss this further on https://github.com/openstack-k8s-operators/openstack-operator/pull/416#discussion_r1294803990